gravitee-apim-console-webui/src/management/api/portal/userGroupAccess/groups/api-portal-access-groups.component.spec.ts

```diff
@@ -0,0 +1,196 @@
+        { provide: UIRouterState, useValue: fakeAjsState },
+        { provide: UIRouterStateParams, useValue: { apiId: API_ID } },
+        { provide: CurrentUserService, useValue: { currentUser } },
+      ],
+    })
+      .overrideProvider(InteractivityChecker, {
+        useValue: {
+          isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        },
+      })
+      .compileComponents();
+
+    fixture = TestBed.createComponent(ApiPortalAccessGroupsComponent);
+    loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+
+    fixture.detectChanges();
+  };
+
+  afterEach(() => {
+    httpTestingController.verify();
+  });
+
+  describe('Groups tab for user with writing rights', () => {
+    beforeEach(async () => await init(['api-definition-u', 'api-gateway_definition-u']));
+
+    it('should show groups', async () => {
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID }));
+
+      const fillForm = await loader.getHarness(MatFormFieldHarness);
+      fillForm.getLabel().then((label) => expect(label).toEqual('Groups'));
+
+      const selection = await loader.getHarness(MatSelectHarness);
+      await selection.open();
+
+      await selection.clickOptions({ text: 'My Group' });
+
+      expect(await selection.getValueText()).toEqual('My Group');
+      expect(fixture.nativeElement.innerHTML.includes("Your API's groups")).toEqual(false);
+    });
+
+    it('should pre-select groups found in user + save new groups', async () => {
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' }), fakeGroup({ id: 'new_group', name: 'New Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group'] }));
+      const fillForm = await loader.getHarness(MatFormFieldHarness);
+      expect(await fillForm.isControlDirty()).toEqual(false);
+
+      const selection = await loader.getHarness(MatSelectHarness);
+      await selection.open();
+
+      const allOptions = await selection.getOptions();
+      expect(allOptions.length).toEqual(2);
+
+      const selectedOptions = await selection.getOptions({ isSelected: true });
+      expect(selectedOptions.length).toEqual(1);
+      expect(await selectedOptions[0].getText()).toEqual('My Group');
+
+      await selection.clickOptions({ text: 'New Group' });
+      expect(await selection.getValueText()).toEqual('My Group, New Group');
+      await selection.close();
+
+      expect(await fillForm.isControlDirty()).toEqual(true);
+
+      const saveBar = await loader.getHarness(GioSaveBarHarness);
+      expect(await saveBar.isVisible()).toEqual(true);
+      await saveBar.clickSubmit();
+
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group'] }));
+      expectOneApiPut(fakeApi({ id: API_ID, groups: ['my_group', 'new_group'] }));
+
+      // expect reloaded component
```

Comment:
Personally I don't like this part. because it is already done at the beginning of the test. 
I don't do that: expectOneApiPut but i add directly the POST expectOne here.

moreover you have not tested what you send to the server. i mean the post body. 

what do you think about :
```
const req = httpTestingController.expectOne({ url: `${CONSTANTS_TESTING.env.baseURL}/apis/${api.id}`, method: 'PUT' })
expect(req.body.group).toEqual(NEW STUFF)
```
and not call the fetch to stop the test here 🤷‍♂️